Bug: yen key is mapped to backslash

### Requirements

- [X] I've searched [issues](https://github.com/jtroo/kanata/issues) to see if this has not been reported before.

### Describe the bug

for some reason, kanata maps yen symbol to backslash, which renders my config unusable, because on my keyboard these are two different keys.

https://github.com/jtroo/kanata/blob/b12b6eb9ea359f3de9cbc6e29333d3ca3ebd70e6/parser/src/keys/mod.rs#L112

This is what is generated when I hit "yen"
```
Event: time 1691323054.832263, type 4 (EV_MSC), code 4 (MSC_SCAN), value 70089
Event: time 1691323054.832263, type 1 (EV_KEY), code 124 (KEY_YEN), value 0
```

And this is what for "backslash"
```
Event: time 1691323091.521054, type 4 (EV_MSC), code 4 (MSC_SCAN), value 70032
Event: time 1691323091.521054, type 1 (EV_KEY), code 43 (KEY_BACKSLASH), value 0
```

### Relevant kanata config

```
(defcfg
  linux-dev /dev/input/by-id/usb-TypeMatrix.com_USB_Keyboard-event-kbd
  delegate-to-first-layer yes
  process-unmapped-keys no
  danger-enable-cmd no
)

(defsrc
  esc    f1     f2     f3     f4     f5     del     f6     f7     f8     f9     f10     f11   f12   
  grv    1      2      3      4      5      bspc    6      7      8      9      0       -     =      yen
  tab    q      w      e      r      t              y      u      i      o      p       [     ]      \
  lsft   a      s      d      f      g      ret     h      j      k      l      ;       '     rsft   caps
         z      x      c      v      b              n      m      ,      .      /       ro           hiragana
  lctl   pp     cmp    muhenkan             spc                   henkan home   up      end   rctl   pgup
         lmet   lalt                                              ralt   left   down    rght         pgdn
)

(defalias
    +esc (tap-hold-next-release 200 esc (layer-toggle symbols))
    +tab (tap-hold-next-release 200 tab (layer-toggle numbers))
    +spc (tap-next-release spc (layer-toggle navigation))
)

;;
;; Layers
;;
(deflayer qwerty
  _      _      _      _      _      _       caps    _          _      _      _      _       _     _
  _      q      w      e      r      t       bspc    y          u      i      o      p       [     _      _
  _      a      s      d      f      g               h          j      k      l      ;       '     _      _
  _      z      x      c      v      b       ret     n          m      ,      .      /       \     _      _
         lctl   lmet   lalt   lsft   @+tab           @+esc      rsft   `      -      =       ]            _
  _      _      _      _                     @+spc              _      _      _      _       _     _
         _      _                                               _      _      _      _             _
)

(deflayer navigation
  _      _      _      _      _      _      mute    _      _      _      _      _       _     _
  _      f1     f2     f3     f4     f5     volu    _      _      _      _      _       _     _      _
  _      f6     f7     f8     f9     f10            left   down   up     rght   _       _     _      _
  _      _      _      _      _      _      vold    _      _      _      _      _       _     _      _
         _      _      _      _      _              _      _      _      _      _       _            _
  _      _      _      _                    _                     _      _      _       _     _      _
         _      _                                                 _      _      _       _            _
)

(deflayer numbers
  _      _      _      _      _      _       del     _          _      _      _      _       _     _
  _      _      _      _      _      _       _       _          _      7      8      9       /     _      _
  _      _      _      _      _      _               _          _      4      5      6       *     _      _
  _      _      _      _      _      _       _       _          _      1      2      3       -     _      _
         _      _      _      _      _               _          _      0      .      =       +            _
  _      _      _      _                     _                  _      _      _       _      _     _
         _      _                                               _      _      _       _            _
)

(deflayer symbols
  _      _      _      _      _      _      _       _      _      _      _      _       _     _
  ~      !      @      #      $      %      _       ^      &      *      \(     \)      {     _      _
  _      _      _      _      _      _              -      \_     -      +      =       }     _      _
  _      _      _      _      _      _      _       _      _      _      _      _       _     _      _
         _      _      _      _      _              _      _      _      _      _       _            _
  _      _      _      _                    _                     _      _      _       _     _      _
         _      _                                                 _      _      _       _            _
)

```


### To Reproduce

1. connect typematrix 2030
2. turn it into 106-key mode (fn+f3)
3. try to start kanata with the config


### Expected behavior

kanata should accept config

### Kanata version

kanata 1.4.0

### Debug logs

```
2023-08-06T14:52:07.820326582+03:00 [INFO] kanata v1.4.0 starting
2023-08-06T14:52:07.820738611+03:00 [ERROR]   × Error in configuration file
    ╭─[/etc/kanata/typematrix.kbd:10:1]
 10 │   grv    1      2      3      4      5      bspc    6      7      8      9      0       -     =      yen
 11 │   tab    q      w      e      r      t              y      u      i      o      p       [     ]      \
    ·                                                                                                      ┬
    ·                                                                                                      ╰── Error here
 12 │   lsft   a      s      d      f      g      ret     h      j      k      l      ;       '     rsft   caps
    ╰────
  help: Repeat declaration of key in defsrc: "\"
        
        For more info, see the configuration guide or ask in GitHub discussions.
            guide : https://github.com/jtroo/kanata/blob/main/docs/config.adoc
            ask   : https://github.com/jtroo/kanata/discussions

2023-08-06T14:52:07.820848123+03:00 [ERROR] failed to parse file


Press enter to exit
```

### Operating system

Fedora 38

### Additional context

_No response_

--- a/docs/config.adoc
+++ b/docs/config.adoc
@@ -63,10 +63,11 @@ amount of whitespace (spaces, tabs, newlines) are not relevant. You may use
 spaces, tabs, or newlines however you like to visually format `defsrc` to your
 liking.
 
-The the primary source of all key names is the
-https://github.com/jtroo/kanata/blob/main/parser/src/keys/mod.rs[str_to_oscode]
-function in the source code. Please feel free to file an issue if you're unable
-to find the key you're looking for.
+The the primary source of all key names are the
+`str_to_oscode` and `default_mappings` functions in
+https://github.com/jtroo/kanata/blob/main/parser/src/keys/mod.rs[the source].
+Please feel welcome to file an issue
+if you're unable to find the key you're looking for.
 
 An example `defsrc` containing the US QWERTY keyboard keys as an
 approximately 60% keyboard layout:

--- a/parser/src/keys/mod.rs
+++ b/parser/src/keys/mod.rs
@@ -66,6 +66,10 @@ fn add_default_str_osc_mappings(mapping: &mut HashMap<String, OsCode>) {
         (",", OsCode::KEY_COMMA),
         (".", OsCode::KEY_DOT),
         ("\\", OsCode::KEY_BACKSLASH),
+        // Mapped as backslash because in some locales/fonts, yen=backslash
+        ("yen", OsCode::KEY_BACKSLASH),
+        // Unicode yen is probably the yen key, so map this to a separate oscode by default.
+        ("¥", OsCode::KEY_YEN),
     ];
     for dm in default_mappings {
         mapping.entry(dm.0.into()).or_insert(dm.1);
@@ -109,7 +113,7 @@ pub fn str_to_oscode(s: &str) -> Option<OsCode> {
         "p" => OsCode::KEY_P,
         "lbrc" => OsCode::KEY_LEFTBRACE,
         "rbrc" => OsCode::KEY_RIGHTBRACE,
-        "bksl" | "yen" | "¥" => OsCode::KEY_BACKSLASH,
+        "bksl" => OsCode::KEY_BACKSLASH,
         "caps" => OsCode::KEY_CAPSLOCK,
         "a" => OsCode::KEY_A,
         "s" => OsCode::KEY_S,

